src/tests/loadtxt/CMakeLists.txt

```diff
@@ -1,15 +1,10 @@
-include_directories(${PROJECT_BINARY_DIR}/src)
-
-project(loadtxt)
-
 add_executable(test_loadtxt test_loadtxt.f90)
 target_link_libraries(test_loadtxt fortran_stdlib)
 
 add_executable(test_savetxt test_savetxt.f90)
 target_link_libraries(test_savetxt fortran_stdlib)
 
-add_test(test_loadtxt ${PROJECT_BINARY_DIR}/test_loadtxt)
-add_test(test_savetxt ${PROJECT_BINARY_DIR}/test_savetxt)
-
-file(COPY array1.dat  array2.dat  array3.dat  array4.dat
-     DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+add_test(NAME load_text COMMAND $<TARGET_FILE:test_loadtxt> ${CMAKE_CURRENT_BINARY_DIR}
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+add_test(NAME save_text COMMAND $<TARGET_FILE:test_savetxt> ${CMAKE_CURRENT_BINARY_DIR}
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
```

Comment:
The alternative to this is to use test fixtures to move files around. In addition, care should be taken to prevent issues when testing in parallel if multiple tests read and/or modify the same files, or if one test consumes a file created or modified by another test.

Reply:
(But, just to clarify, this approach makes sense here, as far as I can tell from a very cursory look at the code in the tests.)

Reply:
yes I think so too, the test is rudimentary at the moment and doesn't yet need these protections.